File: examples/Effects.hs
module Main where

import Control.Monad.State

data MyState = MyState Int

mystateAdd :: Int -> StateT MyState IO ()
mystateAdd valueDiff = do
  MyState value <- get
  put (MyState (value + valueDiff))

mystateValue :: StateT MyState IO Int
mystateValue = do
  MyState value <- get
  liftIO $ putStrLn "Value was read!"
  return value

mainAction :: StateT MyState IO ()
mainAction = do
    mystateAdd 3
    value <- mystateValue
    liftIO . putStrLn $ show value

main = runStateT mainAction (MyState 7)

File: examples/effects.py
#!/usr/bin/python

class State:
    def __init__(self,value):
        self._value = value

    def add(self, valueDiff):
        self._value += valueDiff

    @property
    def value(self):
        print "Value was read!"
        return self._value

def main():
    s = State(7)
    s.add(3)
    print s.value

if __name__ == "__main__":
    main()